Update KNX component to xknx 0.11

farmio opened the pull request:
## Breaking Change:

<!-- What is breaking and why we have to break it. Remove this section only if it was NOT a breaking change. -->
- `scene`: scene_number is now 1 indexed according to KNX standards. Previously it was 0 based. Please add 1 to your already configured scene numbers (`scene_number: 5` -> `scene_number: 6`).
- `sensor`: `state_address` replaces `address` in configuration
- `binary_sensor`: `state_address` replace `address` in configuration
- *when using xknx config file* (`knx: config_file = ...`): Replace `group_address` in BinarySensor and Sensor with `group_address_state`.

## Description:

Updates the knx component to use xknx 0.11.1 . This introduces several new features and bugfixes. For a complete list see: https://github.com/XKNX/xknx/releases/tag/0.11.0

The main new features are
- New sensor types including `string`, `scene_number` and DPT8 types
- sensor and binary_sensor now have an option to disable periodic state requests (`sync_state = False` - eg. for actuators that don't support it or sensors of type 'scene_number')
- Binary values are now exposable to the knx bus
- Add support for RGBW lights (DPT 251.600)
- Auto detection of local IP address (so `local_ip` is optional now for tunnelling)

**Related issue (if applicable):** 
fixes #21737 (pulse sensor type fixed)
fixes #22768 (string sensor type is now supported by default)
fixes #19974 (climate devices with setpoint_shift fixed)
closes #22335 (at least I think its closable by now)
closes #18620 (was fixed in last xknx release)


**Pull request with documentation for [home-assistant.io](https://github.com/home-assistant/home-assistant.io) (if applicable):** home-assistant/home-assistant.io#9682

## Example entry for `configuration.yaml` (if applicable):
see home-assistant.io and PR 👆

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] [_The manifest file_][manifest-docs] has all fields filled out correctly. Update and include derived files by running `python3 -m script.hassfest`.
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `python3 -m script.gen_requirements_all`.
  - [x] Untested files have been added to `.coveragerc`. (the knx component was already added)

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

farmio committed:
update component for xknx 0.11.0

- expose sensor state is not casted to float anymore
- climate mode operation list has no more None values
- light supports white_value (rgbw)
- sensor expects `group_address_state` now instead of `group_address`
- sensor forwards device_class if available

farmio committed:
update manifest to use xknx 0.11.0

farmio committed:
update requirements_all for xknx 0.11.0

Julius2342 commented on homeassistant/components/knx/binary_sensor.py:
```diff
@@ -63,24 +63,24 @@ def async_add_entities_discovery(hass, discovery_info, async_add_entities):
         hass.data[DATA_KNX].xknx,
         name=name,
-        group_address=config.get(CONF_ADDRESS),
+        group_address_state=config[CONF_ADDRESS],
```
It is a state address, so i think we should use `address_state` as the name of the value. I know that people have to update their configuration, but this would make the config more consistent.

farmio replied:
same for sensors (not binary_)?

farmio replied:
done for both.

farmio committed:
update for xknx 0.11.1

- require xknx 0.11.1
- use 'state_address' instead of 'address' in sensor and binary_sensor configuration
- optional 'sync_state' for sensors and binary_sensors

farmio committed:
Merge branch 'dev' into knx-xknx0.11

MartinHjelmare commented on homeassistant/components/knx/light.py:
```diff
@@ -275,13 +294,14 @@ def supported_features(self):
 
     async def async_turn_off(self, **kwargs):
         """Turn the light off."""
+        del kwargs
```
Why is this done?

farmio replied:
@phbaer could you have a look, please?

farmio replied:
I have tested it with this line removed, as I have no   idea what it is supposed to do. Everything continues working, but I have no   RGBW lights to test against - this was added with the RGBW feature - https://github.com/XKNX/xknx/pull/206

MartinHjelmare replied:
Just remove it. :+1:

MartinHjelmare commented on homeassistant/components/knx/sensor.py:
```diff
@@ -93,6 +98,11 @@ def unit_of_measurement(self):
+    @property
+    def device_class(self):
+        """Return the device class of the sensor."""
+        return self.device.ha_device_class()
```
Is xknx aware of home assistant sensor device classes?

farmio replied:
Yes, xknx sensor classes are. The device class is returned as a string - or `None` if there is no   fitting HA device class.
https://github.com/XKNX/xknx/blob/cc10a82d11b4a60bd9765a7b410ab05edd5da41f/xknx/devices/remote_value_sensor.py#L121

MartinHjelmare replied:
Ok. Coupling the two packages this tightly is fragile. If the home assistant API would change, something could break until the xknx library is updated. I would recommend to have a dict in this module that maps between the xknx sensor class and the home assistant device class.

But since the xknx library seems to be built for home assistant I don't think it's strictly necessary.

farmio replied:
Xknx works on its own but is built with HA as a main target.
With almost 50 different sensor classes its nice to have everything in one place (range, unit of measurement, ha_device_class, ...) instead of spreading these attributes out to different modules.

We could import the DEVICE_CLASS_* constants in this module and map against our ha_device_class attribute. To me this would feel kind of redundant.

farmio committed:
remove questionable `del kwargs`

farmio committed:
Merge branch 'knx-xknx0.11' of https://github.com/farmio/home-assistant into knx-xknx0.11

farmio committed:
Merge remote-tracking branch 'upstream/dev' into knx-xknx0.11

farmio commented:
Updated form home-assistant/dev to resolve unrelated test failures.

MartinHjelmare commented:
Great!

MartinHjelmare commented:
@Julius2342 are we good to merge?

Julius2342 commented:
@MartinHjelmare : Yes! 👍

derandiunddasbo commented:
Seems to work fine so far, f.e. binary sensor states are finally updated on startup, woohoo! :-) 

...and I can see the states of my DPT 16 sensors without patching xknx myself anymore! 

Speaking of which: As of now, the KNX Sensor Doc (https://www.home-assistant.io/components/sensor.knx) doesn't mention the new DPT's (string, DPT-8, scene_number).

farmio commented:
👍
Documentation is written but it seems its PR ( home-assistant/home-assistant.io#9682 ) was merged to the wrong branch.

frenck commented:
The documentation PR was merged to the correct branch, this went out of sync with the cut-offs and overseen, correcting that.
